src/ProducerSideService/BuildAndStartProducerSideServerLinux.cpp

```diff
@@ -3,40 +3,75 @@
 // found in the LICENSE file.
 
 #include <absl/strings/str_format.h>
+#include <sys/socket.h>
 #include <sys/stat.h>
+#include <sys/un.h>
 
 #include <filesystem>
+#include <memory>
 #include <system_error>
 
 #include "BuildAndStartProducerSideServerWithUri.h"
+#include "OrbitBase/Result.h"
 #include "OrbitBase/SafeStrerror.h"
+#include "ProducerEventProcessor/ProducerEventProcessor.h"
 #include "ProducerSideChannel/ProducerSideChannel.h"
 #include "ProducerSideService/BuildAndStartProducerSideServer.h"
+#include "ProducerSideService/ProducerSideServer.h"
 
 namespace orbit_producer_side_service {
 
-std::unique_ptr<ProducerSideServer> BuildAndStartProducerSideServer() {
+// Tries to connect to the unix socket associated with the given path. If the connection succeeds we
+// know that something is already listening there.
+static ErrorMessageOr<void> VerifySocketAvailability(std::string_view socket_path) {
+  int socket_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+
+  sockaddr_un addr{};
+  std::memset(&addr, 0, sizeof(addr));
+  addr.sun_family = AF_UNIX;
+  socket_path.copy(addr.sun_path, sizeof(addr.sun_path) - 1);
+
+  if (connect(socket_fd, reinterpret_cast<sockaddr*>(&addr), sizeof(addr)) == 0) {
+    close(socket_fd);
+
+    // TODO(b/229048915): Update the error message once the cloud collector is released.
+    return ErrorMessage{"Orbit Service is already running on the instance."};
```

Comment:
```suggestion
    return ErrorMessage{"OrbitService is already running on the instance."};
```
I guess we don't usually add the space to the name?